core/integration/bookkeeping/plugin.go

```diff
@@ -1246,6 +1248,163 @@ func (p *Plugin) CallStack(data interface{}) (stack map[string]interface{}) {
+		runFetchRequest := bkpb.RunFetchRequest{RunNumber: runNumber}
+
+		timeout := callable.AcquireTimeout(BKP_FILL_TIMEOUT, varStack, "RetrieveFillInfo", envId)
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		log.WithField("partition", envId).
+			WithField("level", infologger.IL_Devel).
+			WithField("endpoint", viper.GetString("bookkeepingBaseUri")).
+			WithField("call", "RetrieveFillInfo").
+			Debugf("requesting BK the run info for run %d", runNumber)
+		runWithRelations, err := p.bookkeepingClient.RunServiceClient.Get(ctx, &runFetchRequest, grpc.EmptyCallOption{})
+		if err != nil {
+			return nil, err
+		}
+		lhcFill := runWithRelations.GetLhcFill()
+		if lhcFill == nil {
+			return nil, fmt.Errorf("lhcFill for run %d is nil", runNumber)
+		}
+		return lhcFill, nil
+	}
+
+	fetchLatestLHCInfo := func() (out *bkpb.LHCFill, err error) {
+		lhcFillFetchRequest := bkpb.LastLhcFillFetchRequest{}
+
+		timeout := callable.AcquireTimeout(BKP_FILL_TIMEOUT, varStack, "RetrieveFillInfo", envId)
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		log.WithField("partition", envId).
+			WithField("level", infologger.IL_Devel).
+			WithField("endpoint", viper.GetString("bookkeepingBaseUri")).
+			WithField("call", "RetrieveFillInfo").
+			Debug("requesting BK the latest LHC fill info")
+		LhcFillWithRelations, err := p.bookkeepingClient.LhcFillServiceClient.GetLast(ctx, &lhcFillFetchRequest, grpc.EmptyCallOption{})
+		if err != nil {
+			return nil, err
+		}
+		lhcFill := LhcFillWithRelations.GetLhcFill()
+		if lhcFill == nil {
+			return nil, fmt.Errorf("lhcFill is nil")
+		}
+		return lhcFill, nil
+	}
+
+	propagateLHCInfoToVarStack := func(lhcInfo *bkpb.LHCFill, varStack map[string]string) {
+		call.VarStack["fill_info_fill_number"] = string(lhcInfo.FillNumber)
+		call.VarStack["fill_info_filling_scheme"] = lhcInfo.FillingSchemeName
+		call.VarStack["fill_info_beam_type"] = lhcInfo.BeamType
+		if lhcInfo.StableBeamStart != nil {
+			call.VarStack["fill_info_stable_beam_start_ms"] = strconv.FormatInt(*lhcInfo.StableBeamStart, 10)
+		}
+		if lhcInfo.StableBeamEnd != nil {
+			call.VarStack["fill_info_stable_beam_end_ms"] = strconv.FormatInt(*lhcInfo.StableBeamEnd, 10)
+		}
+		log.WithField("partition", envId).
+			WithField("level", infologger.IL_Devel).
+			WithField("endpoint", viper.GetString("bookkeepingBaseUri")).
+			WithField("call", "RetrieveFillInfo").
+			Infof("successfully updated the LHC Fill info (number %d, scheme %s, beam type %s)",
+				lhcInfo.FillNumber, lhcInfo.FillingSchemeName, lhcInfo.BeamType)
+	}
+	deleteLHCInfoInVarStack := func(varStack map[string]string) {
+		delete(call.VarStack, "fill_info_fill_number")
+		delete(call.VarStack, "fill_info_filling_scheme")
+		delete(call.VarStack, "fill_info_beam_type")
+		delete(call.VarStack, "fill_info_stable_beam_start_ms")
+		delete(call.VarStack, "fill_info_stable_beam_end_ms")
+	}
```

Comment:
This operates on the varStack of the callRole i.e. the immediate context, but doesn't affect the varStack of the environment as a whole beyond that specific role. I propose that you use `ParentRole.SetGlobalRuntimeVar` and `ParentRole.DeleteGlobalRuntimeVar` instead, which operate on the varStack of the root role and therefore make the values available throughout the environment.

For an example, see [this usage in the TRG plugin](https://github.com/AliceO2Group/Control/blob/54ac1915c3d477fea40b2709241f4e14b490244e/core/integration/trg/plugin.go#L667C28-L667C28).